test/feedsme.test.js

```diff
@@ -462,167 +483,282 @@ describe('feedsme', function () {
-
-        root = increment(root, 'payload', { env, inc });
-        rootPackage = increment(rootPackage, 'package', { env, inc });
-        rootVersion = increment(rootVersion, 'version', { env, inc });
-
-        await Promise.all([
-          Package.create(rootPackage),
-          Version.create(rootVersion)
-        ]);
-
-        await change(env, root);
-        assume(logSpy.args[3]).contains(`Not triggering dependent build for huh@2.0.0, doesnt include what version 3.0.0`);
-        sinon.restore();
-
-        //
-        // Bump child package and dependencies to update release line
-        //
-        child = increment(child, 'payload', { env, inc, dependencies: [root.name] });
-        childPackage = increment(childPackage, 'package', { env, inc, dependencies: [root.name] });
-        childVersion = increment(childVersion, 'version', { env, inc });
-        // Manually publish so the release line gets
-        await Promise.all([
-          Package.create(childPackage),
-          Version.create(childVersion)
-        ]);
-        mockRequests(childPackage.name, childPackage.version);
-        await change(env, child);
-
-        const release = await fme.release.get({ pkg: rootPackage.name });
-        assume(release.version).equals(rootPackage.version);
-        assume(release.pkg).equals(rootPackage.name);
-        assume(release.dependents).contains(childPackage.name);
-        assume(release.dependents[childPackage.name]).equals(childPackage.version);
-      });
-
-      it('should simulate publish of existing root package (previous major) and build proper dependent package based on previous', async function () {
-        const { Package, Version } = fme.models;
-        let { root, rootPackage, rootVersion, childPackage } = clone(fixtures.first);
-        const env = 'dev';
-        const inc = 'minor';
-
-        root = increment(root, 'payload', { env, inc });
-        rootVersion = increment(rootVersion, 'version', { env, inc });
-        rootPackage = increment(rootPackage, 'package', { env, inc });
-
-        mockRequests(childPackage.name, childPackage.version);
-        await Promise.all([
-          Package.create(rootPackage),
-          Version.create(rootVersion)
-        ]);
-
-        const prevrelease = await fme.release.get({ pkg: rootPackage.name });
-        const [{ body }] = await Promise.all([
-          waitCarpenter('publish'),
-          change(env, root)
-        ]);
-
-        const latest = fme.extractLatest(body);
-        delete latest._id;
-        assume(semver.satisfies(root.version, latest.dependencies[rootPackage.name]));
-
-        await Promise.all([
-          Package.create(latest),
-          Version.create({
-            versionId: `${latest.name}@${latest.version}`,
-            name: latest.name,
-            version: latest.version,
-            value: JSON.stringify(body)
-          })
-        ]);
+      describe('sequence with huh and what packages', function () {
```

Comment:
We **really** need better naming conventions and documentation around these scenarios. I believe the best path forward is to start making true [sequence diagrams]() for these scenarios. UML isn't always the answer, but sequence diagrams were designed for communicating exactly these kinds of time-based edge cases. 

If we had a sequence diagram for each `describe` we would have a high level of confidence in our test coverage. Reducing the unknown unknowns.

Reply:
Agreed.